run_VAE.py

```diff
@@ -1,62 +1,126 @@
+        if not arguments_.supplementary:
+            raise AttributeError("supplementary directory must be specified when method = assemble")
+
+    # process needs raw (load _file_paths), and target (torch weights)
+    elif arguments_.method == 'process':
+        if not arguments_.raw:
+            raise AttributeError("raw directory must be specified when method = process")
+        if not arguments_.weights:
+            raise AttributeError("pytorch VQ-VAE weights path must be specified when method = process")
+
+    # trajectory matching needs raw (load file_paths, write trajectories), supp (load cell_traj)
+    elif arguments_.method == 'trajectory_matching':
+        if not arguments_.raw:
+            raise AttributeError("raw directory must be specified when method = assemble")
+        if not arguments_.supplementary:
+            raise AttributeError("supplementary directory must be specified when method = assemble")
+
+    if arguments_.fov:
+        sites = arguments_.fov
+    else:
+        # get all "XX-SITE_#" identifiers in raw data directory
+        sites = [os.path.splitext(site)[0][0:9].split('_NN')[0] for site in os.listdir(inputs) if
+                 site.endswith(".npy")]
+        sites = list(set(sites))
 
-def main():
-    sites = sites_NOVEMBER
-    inputs = RAW_NOVEMBER
-    outputs = INTERMEDIATE_NOVEMBER
-    
     wells = set(s[:2] for s in sites)
-    process = []
     for i, well in enumerate(wells):
         well_sites = [s for s in sites if s[:2] == well]
         print(well_sites)
-        args = (inputs, outputs, TARGET, well_sites)
-        p = Worker(args, gpuid=i)
+        # if method == "assemble":
+        #     # for "assemble" it is coded such that first arg is the output directory, second arg is input
+        #     args = (outputs, inputs, TARGET, well_sites)
+        if method == "process":
+            if arguments_.weights is None:
+                raise AttributeError("path to VQ-VAE weights must be defined for method=process")
+            else:
+                weights = arguments_.weights
+                args = (inputs, None, weights, well_sites)
+        else:
+            args = (inputs, outputs, None, well_sites)
+        p = Worker(args, gpuid=i, method=method)
         p.start()
         p.join()
 
+
+def parse_args():
+    """
+    Parse command line arguments for CLI.
+
+    :return: namespace containing the arguments passed.
+    """
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        '-i', '--raw',
+        type=str,
+        required=False,
+        help="Path to multipage-tiff file of format [t, x, y]",
+    )
+    parser.add_argument(
+        '-o', '--supplementary',
+        type=str,
+        required=False,
+        help="Path to write results",
+    )
+    parser.add_argument(
+        '-m', '--method',
+        type=str,
+        required=True,
+        choices=['assemble', 'process', 'trajectory_matching'],
```

Comment:
Do you think we should keep it as a choice out of three or any allowed combination (in which case the three procedures could be called in one run)?